lebedigital/openbis/expstep.py

```diff
@@ -0,0 +1,1464 @@
+# -*- coding: utf-8 -*-
+
+from math import isnan
+from pathlib import Path
+from pybis import Openbis
+import pandas as pd
+import json
+import os
+import shutil
+import logging
+from getpass import getpass
+import sys
+from sys import exit
+import yaml
+from termcolor import colored
+
+
+class ExpStep:
+    def __init__(
+            self,
+            name: str = '',
+            type: str = '',
+            metadata: dict = None,
+            space: str = "",
+            project: str = "",
+            collection: str = "",
+            parents: list = None,
+            identifier: str = '',
+            permId: str = '',
+            sample_object=None,
+            datasets: list = None,
+            dataset_codes: list = None,
+    ) -> None:
+        """Creates an ExpStep Object
+
+        Args:
+            name (str, optional): name of the experiment. Defaults to ''.
+            type (str, optional): type of the experiment, has to be a defined sample type in the Datastore. Defaults to ''.
+            metadata (dict, optional): metadata of your data, is defined by the sample type. Defaults to None.
+            space (str, optional): space in which the experiment should be saved. Defaults to "".
+            project (str, optional): project in which the experiment should be saved. Defaults to "".
+            collection (str, optional): collection in which the expeiment should be saved.. Defaults to "".
```

Comment:
That is something i would like to do later. I will want to write the docs using sphinx probably and have some explanations/usecases for the users to see. I wanted to get the pull request approved first though because the changes were large enough as they are for me.